### PR TITLE
Js cleanup

### PR DESCRIPTION
**Pull request by maxkorp**
This is best reviewed commit by commit. Doesn't do anything functionally, but trying to get the code in order a bit, it's been all over the place for quite a while.

### CODE REVIEW

File: lib/reference.js [hunk 1]
```diff
@@ -5,39 +5,39 @@ var Reference = NodeGit.Reference;
 var Branch = NodeGit.Branch;
 
 /**
-* Retrieves the reference pointed to by the oid
+* Retrieves the reference by it's short name
```
Comment (implausible) [inline lib/reference.js]: `its`

Comment (maxkorp) [inline lib/reference.js]: Fair, but thats not new, it's preexisting (just looks new because of how the diff is shown)

File: lib/repository.js [hunk 2]
```diff
@@ -460,155 +862,117 @@ Repository.prototype.getHeadCommit = function(callback) {
...
+ * @return {Array<String>}
  */
-Repository.prototype.treeBuilder = function() {
-  var builder = TreeBuilder.create(null);
-
-  builder.root = builder;
-  builder.repo = this;
-
-  return builder;
+Repository.prototype.getReferenceNames = function(type, callback) {
```
Comment (implausible) [inline lib/repository.js]: swap with getReferences

File: lib/repository.js [hunk 3]
```diff
@@ -460,155 +862,117 @@ Repository.prototype.getHeadCommit = function(callback) {
...
- * @param {Buffer} buffer
- * @return {Oid}
+ * @async
+ * @param {Reference.TYPE} type Type of reference to look up
+ * @return {Array<Reference>}
  */
-Repository.prototype.createBlobFromBuffer = function(buffer, callback) {
-  return Blob.createFromBuffer(this, buffer, buffer.length, callback);
+Repository.prototype.getReferences = function(type, callback) {
+  return Repository.getReferences(this, type, false, callback);
```
Comment (implausible) [inline lib/repository.js]: swap with getReferenceNames

File: lib/tree_entry.js [hunk 4]
```diff
@@ -89,6 +85,10 @@ TreeEntry.prototype.toString = function() {
   return this.path();
 };
 
-TreeEntry.prototype.oid = function() {
+/**
+ * Retrieve the SHA for this TreeEntry.
+ * @return {String}
+ */
+TreeEntry.prototype.sha = function() {
```
Comment (implausible) [inline lib/tree_entry.js]: sha before toString

File: lib/odb.js [hunk 5]
```diff
@@ -1,10 +1,11 @@
 var NodeGit = require("../");
 
 var Odb = NodeGit.Odb;
-var read = Odb.prototype.read;
 
-Odb.prototype.read = function(oid, callback) {
-  return read.call(this, oid).then(function(odbObject) {
+var _read = Odb.prototype.read;
+
+Odb.prototype._read = function(oid, callback) {
```
Comment (implausible) [inline lib/odb.js]: should probably be `Odb.prototype.read`

**Comment by maxkorp** [comment]
Well, fingers crossed!

**Pull request merged**